### ISSUE TITLE
How to pass 'debug' flag to node dev server?

### ISSUE THREAD

**Issue by jcheroske** [zeit/next.js#2044]
Trying to start the node debugger. Tried `next debug` and `next dev debug` in hopes that would get passed on to node, but no luck.

**Comment by arunoda** [comment]
See: https://github.com/zeit/next.js/tree/master#customizing-webpack-config

You can customize the webpack dev middleware. Use the latest stable version of Next.js

**Comment by jcheroske** [comment]
This is what I am currently using now, saved in `package.json` as a script:

`"debug": "node --inspect-brk ./node_modules/next/dist/bin/next-dev",`

**Comment by chmartinez** [comment]
Hi! Newcomer here.
@arunoda I was trying to setup the middleware but I'm kinda lost. The example provided in next's documentation uses a BundleAnalyzer package but I couldn't find a package that "activates" the server side debugging. Maybe it's because I'm not familiar to Webpack but I think that it could be helpful if you could give us one more hint around this.
Thanks!
Christian

PS: I want to be able to debug the "getInitialProps" and all the code that is executed in server side.

**Comment by nwshane** [comment]
Hi @chmartinez, did you figure this out? I don't know if it's possible to debug using the `next` command line tool, but I've done things this way:

1. Add `debugger` to `getInitialProps`
2. Setup a custom server on the backend so that you can call `node server.js` instead of `next start` (explained [here](https://github.com/zeit/next.js#custom-server-and-routing))
3. Run `node --inspect server.js` (I think you need node version >= 8.0.0)
4. If in chrome, open the node debugger: go to about:inspect and click "Open dedicated DevTools for Node"
5. Load the page with debugger in `getInitialProps` - it should pause execution for you to debug

**Comment by chmartinez** [comment]
Hi @nwshane.
Yes, I did exactly what you're mentioning and yes, I was able to debug the server side execution.
However, I had a similar issue to the stuff mentioned at https://github.com/zeit/next.js/issues/2474.
TL;DR: next does not generate sourcemaps for server side execution, so it's kinda hard to follow the execution once you hit a "debugger" breakpoint.
Apparently, this PR (https://github.com/zeit/next.js/pull/2508) should fix that. 
Hope it helps! 
Cheers
